tests/store/test_rest_store.py

```diff
@@ -241,6 +247,66 @@ def test_requestor(self, request):
                                   message_to_json(expected_message))
             assert result.token == "67890fghij"
 
+        with mock.patch('mlflow.store.rest_store.http_request') as mock_http:
+            response = mock.MagicMock
+            experiment = Experiment(
+                experiment_id="123", name="abc", artifact_location="/abc",
+                lifecycle_stage=LifecycleStage.ACTIVE)
+            response.text = json.dumps({
+                "experiment": json.loads(message_to_json(experiment.to_proto()))})
+            mock_http.return_value = response
+            result = store.get_experiment_by_name("abc")
+            expected_message0 = GetExperimentByName(experiment_name="abc")
+            self._verify_requests(mock_http, creds,
+                                  "experiments/get-by-name", "GET",
+                                  message_to_json(expected_message0))
+            assert result.experiment_id == experiment.experiment_id
+            assert result.name == experiment.name
+            assert result.artifact_location == experiment.artifact_location
+            assert result.lifecycle_stage == experiment.lifecycle_stage
+            # Test GetExperimentByName against nonexistent experiment
+            mock_http.reset_mock()
+            nonexistent_exp_response = mock.MagicMock
+            nonexistent_exp_response.status_code = 404
+            nonexistent_exp_response.text =\
+                MlflowException("Exp doesn't exist!", RESOURCE_DOES_NOT_EXIST).serialize_as_json()
+            mock_http.return_value = nonexistent_exp_response
+            assert store.get_experiment_by_name("nonexistent-experiment") is None
+            expected_message1 = GetExperimentByName(experiment_name="nonexistent-experiment")
+            self._verify_requests(mock_http, creds,
+                                  "experiments/get-by-name", "GET",
+                                  message_to_json(expected_message1))
+            assert mock_http.call_count == 1
```

Comment:
Here we confirm that the client doesn't try to fall back to ListExperiments if it gets a RESOURCE_DOES_NOT_EXIST from the server